src/components/canvas/DisplayObject/index.ts

```diff
@@ -0,0 +1,22 @@
+export interface IProps {
+    id?: string
+}
+
+abstract class DisplayObject<T extends IProps> {
```

Comment:
В целом, для дженерик-параметров лучше давать понятные имена с префиксом `T`, чтобы отличать имя параметра от имени конкретного типа.
Конкретно в этом классе нет проблем с именем `T`, но в более сложном коде будет легко запутаться

```suggestion
abstract class DisplayObject<TProps extends IProps> {
```

Reply:
для меня такой дженерик читается хуже, его можно спутать с каким-нибудь типом (например `export type TProps = IProps & { author: User }`), когда же я вижу одну заглавную букву `T` - я четко понимаю, что это не какой-то конкретный тип, а дженерик

Reply:
Как раз для типов префиксы не используют. Общее соглашение использовать префикс `T` именно для дженерик-параметров.
По однобуквенным именам очень сложно понять что именно за параметр тут ожидается. Буквально недавно у нас в коде в десятке классов менял `T` на осмысленное имя, когда проследил логику и увидел, что это один и тот же параметр перетекает из класса в класс.
А если параметров больше одного, то становится еще проще запутаться

Reply:
ок, а что можно использовать в качестве префикса для типов? хочется сразу понимать, что это тип, а не какой-то класс (потому что тип это штука в каком-то смысле виртуальная и хотелось бы ее легко визуально отделять от реальных вещей в коде)

Reply:
возможно постфикс `Type` можно использовать? Например `PropsType`, `GameConfigType`